activation/handler.go

```diff
@@ -356,6 +360,22 @@ func (h *Handler) ContextuallyValidateAtx(atx *types.VerifiedActivationTx) error
 	return err
 }
 
+func (h *Handler) cacheAtx(ctx context.Context, atx *types.ActivationTxHeader) {
+	if !h.cache.IsEvicted(atx.TargetEpoch()) {
```

Comment:
i still think it's cleaner to encapsulate the interaction with db in the cache itself.

the whole point of cache is to reduce contention on database and improve latency and i/o. cache and db should be considered as a single component. it's up to the consensus cache to determine whether it will use a database or not.

it's nicer for other parts of the codebase to just tell cache to handle this atx without accessing db separately.

Reply:
i can rename it from cache to consensus_data

>the whole point of cache is to reduce contention on database and improve latency and i/o. cache and db should be considered as a single component. it's up to the consensus cache to determine whether it will use a database or not.

no this is not the idea. the idea is to store small amount of data that is needed for consensus that eliminates any database access